samples/mqtt/x509_pub_sub/main.cpp

```diff
@@ -0,0 +1,366 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/Api.h>
+#include <aws/crt/StlAllocator.h>
+#include <aws/crt/auth/Credentials.h>
+#include <aws/crt/io/TlsOptions.h>
+
+#include <aws/iot/MqttClient.h>
+
+#include <algorithm>
+#include <aws/crt/UUID.h>
+#include <chrono>
+#include <mutex>
+
+#include "../../utils/CommandLineUtils.h"
+
+using namespace Aws::Crt;
+
+int main(int argc, char *argv[])
+{
+
+    /************************ Setup the Lib ****************************/
+    /*
+     * Do the global initialization for the API.
+     */
+    ApiHandle apiHandle;
+    uint16_t proxyPort(8080);
+    uint32_t messageCount = 10;
+
+    /*********************** Parse Arguments ***************************/
+    Utils::CommandLineUtils cmdUtils = Utils::CommandLineUtils();
+    cmdUtils.RegisterProgramName("basic_pub_sub");
+    cmdUtils.AddCommonMQTTCommands();
+    cmdUtils.RemoveCommand("cert");
+    cmdUtils.RemoveCommand("key");
+    cmdUtils.AddCommonProxyCommands();
+    cmdUtils.AddCommonTopicMessageCommands();
+    cmdUtils.AddCommonX509Commands();
+    cmdUtils.RemoveCommand("x509");
```

Comment:
why does the `x509` option even exist, if the samples are for specific connection types